onnxruntime/core/mlas/lib/amd64/ErfKernelFma3.asm

```diff
@@ -0,0 +1,567 @@
+;
+
+ErfKernelFrame STRUCT
+        ErfBuffer0 OWORD 8 DUP(?)
+        ErfBuffer1 OWORD 8 DUP(?)
+        SavedXmm6 OWORD ?
+        SavedXmm7 OWORD ?
+        SavedXmm8 OWORD ?
+        SavedXmm9 OWORD ?
+        SavedXmm10 OWORD ?
+        SavedXmm11 OWORD ?
+        SavedXmm12 OWORD ?
+        SavedXmm13 OWORD ?
+        SavedXmm14 OWORD ?
+        SavedXmm15 OWORD ?
+        Padding0 QWORD ?
+        Padding1 QWORD ?
+        CountN QWORD ?
+        ReturnAddress QWORD ?
+        PreviousP1Home QWORD ?
+        PreviousP2Home QWORD ?
+        PreviousP3Home QWORD ?
+        PreviousP4Home QWORD ?
+
+ErfKernelFrame ENDS
+
+;++
+;
+; Routine Description:
+;
+;   This routine implements a vectorized kernel for the error
+;   function.
+;
+; Arguments:
+;
+;   Input (rcx) - Supplies the input buffer.
+;
+;   Output (rdx) - Supplies the output buffer.
+;
+;   N (r8)  - Supplies the number of elements to process.
+;
+; Return Value:
+;
+;   None.
+;
+;--
+
+        NESTED_ENTRY MlasErfKernelFma3, _TEXT
+
+        alloc_stack (ErfKernelFrame.ReturnAddress)
+
+        save_xmm128_avx xmm6,ErfKernelFrame.SavedXmm6
+        save_xmm128_avx xmm7,ErfKernelFrame.SavedXmm7
+        save_xmm128_avx xmm8,ErfKernelFrame.SavedXmm8
+        save_xmm128_avx xmm9,ErfKernelFrame.SavedXmm9
+        save_xmm128_avx xmm10,ErfKernelFrame.SavedXmm10
+        save_xmm128_avx xmm11,ErfKernelFrame.SavedXmm11
+        save_xmm128_avx xmm12,ErfKernelFrame.SavedXmm12
+        save_xmm128_avx xmm13,ErfKernelFrame.SavedXmm13
+        save_xmm128_avx xmm14,ErfKernelFrame.SavedXmm14
+        save_xmm128_avx xmm15,ErfKernelFrame.SavedXmm15
+
+        END_PROLOGUE
+
+        lea     rax,MlasErfConstants
+
+        sub     r8,8*4
+        jb      LErfProcessRemainingCount
+
+LComputeErf4x8Loop:
+        vbroadcastss ymm15,ErfConstants.ErfNegZero[rax]
+        vmovups      ymm0,YMMWORD PTR [rcx]        ; original input vx0
+        vmovups      ymm1,YMMWORD PTR 32[rcx]      ; original input vx1
+        vmovups      ymm2,YMMWORD PTR 64[rcx]      ; original input vx2
+        vmovups      ymm3,YMMWORD PTR 96[rcx]      ; original input vx3
+
+        vbroadcastss ymm14,ErfConstants.ErfMaxInt[rax]
+
+        vandps       ymm4,ymm0,ymm15               ; vsign0
+        vandps       ymm5,ymm1,ymm15               ; vsign1
+        vandps       ymm6,ymm2,ymm15               ; vsign2
+        vandps       ymm7,ymm3,ymm15               ; vsign3
+
+        vbroadcastss ymm8,ErfConstants.ErfSMALL_P0[rax]
+        vmovups      YMMWORD PTR ErfKernelFrame.ErfBuffer0[rsp],ymm4
+        vmovups      YMMWORD PTR 32+ErfKernelFrame.ErfBuffer0[rsp],ymm5
+        vmovups      YMMWORD PTR 64+ErfKernelFrame.ErfBuffer0[rsp],ymm6
+        vmovups      YMMWORD PTR 96+ErfKernelFrame.ErfBuffer0[rsp],ymm7
+
+        vmovaps      ymm9,ymm8
+        vmovaps      ymm10,ymm8
+        vmovaps      ymm11,ymm8
+
+        vandps       ymm0,ymm0,ymm14               ; abs(vx0)  va0
```

Comment:
Shouldn't there be logic to use ErfUpperAbsRange?

Reply:
Yes, it it is missing. Add them back.